lib/resources/docker.rb

```diff
@@ -0,0 +1,341 @@
+        j = j.map {|k, v|
+          [k.downcase, v]
+        }.to_h
+        ps.push(j)
+      }
+      ps
+    end
+
+    def parse_images
+      # docker does not support the `json .` function here, therefore we need to emulate that behavior.
+      raw_images = inspec.command('docker images -a --no-trunc --format \'{ "id": {{json .ID}}, "repository": {{json .Repository}}, "tag": {{json .Tag}}, "size": {{json .Size}}, "digest": {{json .Digest}}, "createdat": {{json .CreatedAt}}, "createdsize": {{json .CreatedSince}} }\'').stdout
+      c_images = []
+      raw_images.each_line { |entry|
+        c_images.push(JSON.parse(entry))
+      }
+      c_images
+    end
+
+    # checks if the system is Debian 8+, Ubuntu 16.04+ and RHEL 7+
+    def is_systemd_system
+      return true if inspec.os.redhat? && inspec.os.release.to_i >= 7
+      return true if inspec.os.name == 'ubuntu' && inspec.os.release.to_f >= 16.04
+      return true if inspec.os.name == 'debian' && inspec.os.release.to_i >= 8
+      false
+    end
+
+    # returns parsed params
+    def parse_systemd_values(stdout)
+      SimpleConfig.new(
+        stdout,
+        assignment_re: /^\s*([^=]*?)\s*=\s*(.*?)\s*$/,
+        multiple_values: false,
+      ).params
+    end
+  end
+
+  class DockerContainer < Inspec.resource(1)
+    name 'docker_container'
+    desc ''
+    example "
+      describe docker_container('simple_container') do
+        it { should exist }
+        it { should be_running }
+        its('command') { should eq '/bin/sh' }
+      end
+    "
+
+    # TODO: we need to define if a simple string is a ID or NAME
+    # TODO: handle short ids
+    def initialize(opts = {})
+      # fallback for simple strings
+      if opts.is_a?(String)
+        @opts = { name: opts }
+      else
+        @opts = opts
+      end
+    end
+
+    def exist?
+      !id.nil?
+    end
+
+    def id
+      container_info.ids[0] if container_info.entries.length == 1
+    end
+
+    def running?
+      status.downcase.start_with?('up') if container_info.entries.length == 1
+    end
+
+    def status
+      container_info.status[0] if container_info.entries.length == 1
+    end
+
+    def labels
+      container_info.labels[0] if container_info.entries.length == 1
+    end
+
+    def ports
+      container_info.ports[0] if container_info.entries.length == 1
+    end
+
+    def command
+      container_info.commands[0] if container_info.entries.length == 1
+    end
+
+    def image
+      container_info.images[0] if container_info.entries.length == 1
+    end
+
+    def repo
+      image.split(':')[0] unless image.nil?
+    end
+
+    def tag
+      image.split(':')[1] unless image.nil?
+    end
+
+    def to_s
+      'Docker Container'
+    end
+
+    private
+
+    def container_info
+      return @info if defined?(@info)
+      opts = @opts
+      @info = inspec.docker.containers.where { id == opts[:id] || names == opts[:name] }
```

Comment:
No, since docker only exposes `names`, see https://github.com/chef/inspec/pull/1566/files#diff-a405d35eca4d46fa85c19029d6f3b79bR23. Technically multiple names could be attached, but I have not seen that in real world ...